bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/GarbageCollectorThread.java

```diff
@@ -547,6 +549,9 @@ void doCompactEntryLogs(double threshold, long maxTimeMillis) throws EntryLogMet
         MutableLong end = new MutableLong(start);
         MutableLong timeDiff = new MutableLong(0);
 
+        PriorityQueue<EntryLogMetadata> pq = new PriorityQueue<>(Comparator
```

Comment:
I changed this to compact by bucket (from lowest to highest, still prioritizes most compactable entry logs).
As an overhead, it will keep entry log ids in memory. With 50TB disk and 250MB entry log, all entry logs below threshold that's about 200k longs to keep in memory, we should be ok.